crates/core_simd/src/ops/splat.rs

```diff
@@ -0,0 +1,641 @@
+use super::*;
+use crate::simd::Mask;
+
+// LHS splats
+
+macro_rules! simple_lhs_splat {
+    ( impl<const LANES: usize$(, $tvar:ident)?> $op:ident<Simd<$scalar:ty, LANES>> for $lhs:ty {
+        fn $call:ident
+    }) => {
+        impl<$($tvar,)* const LANES: usize> $op<Simd<$scalar, LANES>> for <Simd<$scalar, LANES> as Vector>::Scalar
```

Comment:
Looking at this again, I'm confused why it's not just implemented for the scalar directly.  Inference issues on existing code?

Reply:
For most ops it's just fully parametric. I was considering whether it's worth specializing the div/rem cases for some scalars.

Reply:
Unless you mean why not
```rust
impl<T, const LANES: usize> for Op<<T as Scalar>::Vector> for T
```
and that just kinda falls apart due to failing to actually constrain all the parameters in the right places.

Reply:
I think I'll check out the branch and build the docs to make sure I understand what's going on.

Reply:
To elaborate on "why sometimes specializing splatting":
For most of the RHS Div/Rem impls, specializing on the "div by zero?" check could potentially bring a performance benefit.
For LHS, it only matters for signed ints (which risk overflow... I think we could make our div "wrap" correctly and so drop the `i{N}::MIN / -1` check, but I have to triple-verify we'd be emitting the right LLVMIR for that, as the alternative is UB).

Reply:
I think that entire question complicates this PR so I am going to simply defer it for later.